### ISSUE TITLE
[DataGrid] Starting v5.3.0 till latest version components ErrorOverlay gets wrapped into div with 0px x 0px size 

### ISSUE THREAD

**Issue by IngusSkaistkalns** [mui/mui-x#4026]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

```tsx
<DataGrid
  ...
  components={{
    ErrorOverlay: ({ graphQLErrors, clientErrors, networkError, message }) => (
        <div style={{ background: 'red' }}>Long custom error message to display</div>
    )
  }}
/>
```

will result error component wrapped in additional div with undesired styling `overflow: visible; width: 0px;`

![image](https://user-images.githubusercontent.com/219684/155482336-93787a6b-5210-4def-b6be-c380f3022007.png)
![image](https://user-images.githubusercontent.com/219684/155482490-5bee2b83-afa0-4dfb-96b1-4b35614eca0d.png)

### Expected behavior 🤔

In version 5.2.2 this wrapped div does not appear:

```tsx
<DataGrid
  ...
  components={{
    ErrorOverlay: ({ graphQLErrors, clientErrors, networkError, message }) => (
        <div style={{ background: 'red' }}>Long custom error message to display</div>
    )
  }}
/>
```

does not have that wrapped div with styling that limits size of ErrorOverlay

![image](https://user-images.githubusercontent.com/219684/155483259-46a7adb0-d37b-47c8-851a-4dd2fcc31b50.png)

![image](https://user-images.githubusercontent.com/219684/155483377-a10a3457-5056-4bbf-b295-ee211e680fef.png)

### Steps to reproduce 🕹

Codesandbox - https://codesandbox.io/s/customemptyoverlaygrid-material-demo-forked-e1uww3?file=/demo.tsx

Steps:

Not sure if there anything other than defining custom `ErrorOverlay` between v5.2.2 and up....

### Context 🔦

Custom error message component/overlay

### Your environment 🌎

Not sure if this is environment specific issue.

**Comment by IngusSkaistkalns** [comment]
I also went though CHANGELOG and could not find anything that could lead to such behaviour as desired.

**Comment by flaviendelangle** [comment]
I think #3832 should solve this issue (@cherniavskii)

**Comment by cherniavskii** [comment]
I've updated the issue with a link to Codesandbox.

As a workaround, you can use `GridOverlay` for now: https://codesandbox.io/s/customemptyoverlaygrid-material-demo-forked-6espse?file=/demo.tsx
After #3832 is released, it won't be necessary.

**Comment by cherniavskii** [comment]
Fixed by #3832
